pkg/handlers/generic/lifecycle/utils/utils_integration_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/test/helpers"
+)
+
+var _ = Describe("Namespace", func() {
+	It("creates a new namespace", func(ctx SpecContext) {
+		c, err := helpers.TestEnv.GetK8sClient()
+		Expect(err).To(BeNil())
+
+		namespaceName := "new"
+
+		Expect(EnsureNamespace(ctx, c, namespaceName)).To(Succeed())
+		Expect(c.Delete(ctx, &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		})).To((Succeed()))
```

Comment:
Deleting namespaces in envtest does not work due to missing kube controllers. Let's remove this and ensure we use a new namespace name for each spec.

```suggestion
```

Reply:
Thanks. I think you fixed this in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/559/commits/07543a8204a96403a35bf24bc10cd534642c5d26, which I'll append here,